test/runnable/test17924.d

```diff
@@ -0,0 +1,25 @@
+// PERMUTE_ARGS:
+
+void main()
+{
+    int x = 0;
+    int y = 1;
+    assert((x ?: y) == y);
+    assert((y ?: x) == y);
+
+    int *p;
+    int *u;
```

Comment:
Prefer to use D style declarations instead of C style. That is: `int* a;`.

Reply:
oh, yeah, sorry, my bad

Reply:
fixed.